ChatSecure/Classes/Controllers/OTRDatabaseManager.m

```diff
@@ -214,6 +215,11 @@ - (BOOL)setupYapDatabaseWithName:(NSString *)name
         YapDatabaseSearchResultsView *searchResultsView = [[YapDatabaseSearchResultsView alloc] initWithFullTextSearchName:FTSName parentViewName:AllBuddiesName versionTag:nil options:nil];
         NSString* viewName = [YapDatabaseConstants extensionName:DatabaseExtensionNameBuddySearchResultsViewName];
         [self.database registerExtension:searchResultsView withName:viewName];
+        
+        // Remove old unused objects
+        [self.readWriteDatabaseConnection asyncReadWriteWithBlock:^(YapDatabaseReadWriteTransaction * _Nonnull transaction) {
+            [transaction removeAllObjectsInCollection:OTRXMPPPresenceSubscriptionRequest.collection];
+        }];
```

Comment:
We could also do a read-only check for the numberOfObjectsInCollection OTRXMPPPresenceSubscriptionRequest.collection before issuing the readwrite. Not sure why this takes so long because I figured it would be an O(1) operation with minimal side effects, possibly it triggers some relationship recalculations.